src/edgeworkers/ew-cli-main.ts

```diff
@@ -0,0 +1,328 @@
+      await cliHandler.showEdgeWorkerIdVersionOverview(ewId, { versionId: versionId, showResult: true });
+    } catch (e) {
+      cliUtils.logAndExit(1, e);
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+program
+  .command("upload <edgeworker-identifier>")
+  .description("Creates a new version of a given EdgeWorker Id which includes the code bundle.")
+  .alias("create-version")
+  .option("--bundle <bundlePath>", "Path to bundle file in tgz format")
+  .option("--codeDir <workingDirectory>", "Working directory that includes main.js and bundle.json files")
+  .action(async function (ewId, options) {
+
+    //either bundle or code working directory must be provided or fail
+    if ((!options.bundle && !options.codeDir) || (options.bundle && options.codeDir))
+      cliUtils.logAndExit(1, "ERROR: You must provide the EdgeWorkers bundle tgz (--bundle) OR the working directory for mainjs and manifest file (--codeDir) to create a new version!");
+
+    try {
+      await cliHandler.createNewVersion(ewId, options);
+    } catch (e) {
+      cliUtils.logAndExit(1, e);
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+program
+  .command("download <edgeworker-identifier> <version-identifier>")
+  .description("Download the code bundle of an EdgeWorker version.")
+  .alias("download-version")
+  .option("--downloadPath <downloadPath>", "Path to store downloaded bundle file; defaults to CLI home directory if not provided")
+  .action(async function (ewId, versionId, options) {
+    try {
+      await cliHandler.downloadTarball(ewId, versionId, options.downloadPath);
+    } catch (e) {
+      cliUtils.logAndExit(1, e);
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+program
+  .command("status <edgeworker-identifier>")
+  .description("List Activation status of a given EdgeWorker Id.")
+  .alias("list-activations")
+  .option("--versionId <versionId>", "Version Identifier")
+  .option("--activationId <activationId>", "Activation Identifier")
+  .action(async function (ewId, options) {
+
+    // Do not provide both versionId and activationId
+    if (options.versionId && options.activationId)
+      cliUtils.logAndExit(1, "ERROR: You may not provide both the Version and the Activation identifiers!");
+    try {
+      await cliHandler.showEdgeWorkerActivationOverview(ewId, options);
+    } catch (e) {
+      cliUtils.logAndExit(1, e);
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+program
+  .command("activate <edgeworker-identifier> <network> <version-identifier>")
+  .description("Activate a Version for a given EdgeWorker Id on an Akamai Network")
+  .alias("av")
+  .action(async function (ewId, network, versionId) {
+
+    // Network must use correct keyword STAGING|PRODUCTION
+    if (network.toUpperCase() !== 'STAGING' && network.toUpperCase() !== 'PRODUCTION')
```

Comment:
Would be good consts (and i realize this is probably my bad coding from before lol)